### ISSUE TITLE
[Bug] Inline eslint config errors due to "call to Function() blocked by CSP"

### ISSUE THREAD

**Issue by makyen** [Tampermonkey/tampermonkey#1685]
## Expected Behavior
Use the following inline eslint config:
```
/* eslint "no-multi-spaces": ["warn", { "ignoreEOLComments": true, "exceptions": { "VariableDeclarator": true, "Property": true } }] */
```

This used to work. It was specifically created for use within prior versions of Tampermonkey in order for some test code not to produce eslint errors. It stopped working a couple/few Tampermonkey versions ago. I *am* able to set that configuration for "no-multi-spaces" if I use a Custom Linter Config in TM settings, so it's not that the config doesn't work. I *assume* that something in the Content Security Policy for the editor page was changed and/or eslint changed how it was executing inline module configuration such that it ends up violating the CSP.

IIRC, it started failing after a Tampermonkey update, rather than an browser update.

Having now checked this behavior in Chrome, it's definitely a page CSP issue, which was likely a TM change.

## Actual Behavior
In Firefox, the line is marked as an error with the tooltip:
> eslint: no-multi-spaces - Configuration for rule "no-multi-spaces" is invalid:  
> call to Function() blocked by CSP

In Chrome, the line is marked with the tooltip:
> eslint: no-multi-spaces - Configuration for rule "no-multi-spaces" is invalid:  
> Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self'".

Both Firefox and Chrome show a long error in the console about (only partially copied here):
> Error compiling schema, function code: ...

## Specifications

- Firefox: 108.x and Chrome 109.0.5414.75 (Official Build) (64-bit)
- TM: 4.18.1 (and at least one version earlier)
- OS: Win 10

## Script

Just adding the above eslint inline config line to the default example userscript shows the error once eslint is run:
```js
// ==UserScript==
// @name MyName
// @namespace http://mywebsite.com
// @Version 1.0
// @description try to take over the world!
// @author Me
// @match http://mywebsite.com/*
// @require http://ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js
// @include https://mywebsite.com/*
// @grant none
// ==/UserScript==
/* eslint "no-multi-spaces": ["warn", { "ignoreEOLComments": true, "exceptions": { "VariableDeclarator": true, "Property": true } }] */
```

**Comment by derjanb** [comment]
Workaround for now:

* open an editor
* open dev tools and select console
* execute this: `localStorage.setItem('lint_worker', 'false')`

**Comment by derjanb** [comment]
Duplicate of #1388
